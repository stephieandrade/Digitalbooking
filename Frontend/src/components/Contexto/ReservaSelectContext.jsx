import {createContext} from "react";

const SelectReservaContext = createContext(null);

export default SelectReservaContext 
