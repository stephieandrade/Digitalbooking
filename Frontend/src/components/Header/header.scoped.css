header {
  display: flex;
  justify-content: center;
}

.contenedor-header {
  display: flex;
  justify-content: space-between;
  background-color: var(--five-color);
  align-items: flex-end;
  padding: 7px;
  padding-left: var(--margin-mobile);
  padding-right: var(--margin-mobile);
  align-items: center;
  width: 100%;
  max-width: 1368px;
}

@media screen and (min-width: 768px) {
  .contenedor-header {
    padding-left: var(--margin-tablet);
    padding-right: var(--margin-tablet);
    gap: 10px;
    align-items: flex-end;
  }
}
@media screen and (min-width: 992px) {
  .contenedor-header {
    padding-left: var(--margin-desktop);
    padding-right: var(--margin-desktop);
  }
}
