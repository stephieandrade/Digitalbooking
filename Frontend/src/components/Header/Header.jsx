import React, { useState, useEffect, useContext } from "react";
import Logo from "../Logo/Logo";
import BotonesHeader from "../BotonesHeader/BotonesHeader";
import Avatar from "../Avatar/Avatar";
import Navbar from "../Navbar/Navbar";
import { Contexto } from "../Contexto/Contexto";
import "./header.scoped.css";
import AdminLink from "../AdminLink/AdminLink";

function Header() {
  const [content, setContent] = useState("");
  const { estado } = useContext(Contexto);

  useEffect(() => {
    if (estado.session && JSON.stringify(estado.session) !== "{}") {
      setContent(
        <>
          <Avatar name={estado.session.nombreCompleto} className="" />
          <Navbar />
        </>
      );
    } else {
      setContent(
        <>
          <BotonesHeader />
          <Navbar />
        </>
      );
    }
  }, [estado]);

  return (
    <>
      <header>
        <div className="contenedor-header">
          <Logo />
          {content}
        </div>
      </header>
    </>
  );
}
export default Header;
