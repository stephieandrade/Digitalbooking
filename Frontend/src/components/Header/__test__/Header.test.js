import React from "react";
import { Contexto } from "../../Contexto/Contexto";
import Header from "../Header";
import renderer from 'react-test-renderer';
import { StaticRouter } from 'react-router'


describe("Header", () => {
  it("deber√≠a mostrar <Header /> correctamente ", () => {
    const wrapper = renderer.create(
        <StaticRouter>
      <Contexto.Provider
        value={{
          estado: {
            session: {
              apellido: "diaz",
              nombre: "diego",
              sub: "diego@mail.com",
              nombreCompleto: "Diego Diaz",
            },
          },
        }}
      >
        {" "}
        <Header />{" "}
      </Contexto.Provider></StaticRouter>
    );
    let tree = wrapper.toJSON();
 
    expect(tree).toMatchSnapshot();
  });


  it("deberia tener un div con classname = contenedor-header", () => {
    const wrapper = renderer.create(
        <StaticRouter>
      <Contexto.Provider
        value={{
          estado: {
            session: {
              apellido: "diaz",
              nombre: "diego",
              sub: "diego@mail.com",
              nombreCompleto: "Diego Diaz",
            },
          },
        }}
      >
        {" "}
        <Header />{" "}
      </Contexto.Provider></StaticRouter>
    );
    const testInstance = wrapper.root;
    const element = testInstance.findByType("div")
    expect(element.props.className.includes("contenedor-header")).toBe(true);
  });


});
