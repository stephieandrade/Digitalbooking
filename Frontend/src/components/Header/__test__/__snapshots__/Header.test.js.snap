// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header deber√≠a mostrar <Header /> correctamente  1`] = `
Array [
  " ",
  <header>
    <div
      className="contenedor-header"
    >
      <div
        className="logo"
      >
        <a
          href="/"
          onClick={[Function]}
        >
          <img
            alt="Logo"
            src="logo.png"
          />
        </a>
        <p>
          Rent your car
        </p>
      </div>
      
    </div>
  </header>,
  " ",
]
`;
